profile/profile.go

```diff
@@ -0,0 +1,89 @@
+package profile
+
+import (
+	"fmt"
+	"os"
+)
+
+type Profile interface {
```

Comment:
One more question, more philosophical so I'll leave it up to you. Right now I'm not sure I see the need for a Profile _interface_. I think we could get away with just the UserProfile struct with the methods defined below. It's true that we could keep the interface and add a mock implementation, the way we did with Session. But where would we use that mock implementation? Would it provide anything useful? This package just seems straightforward enough to not require much code 🤔

Reply:
Hmmm, I'd think it might be best as well to stick with just using the struct since I'm finding myself maneuvering a bit to find utility for a mock implementation. I agree that since the methods are mostly managing the struct properties, the interface might be a bit overkill. I can go ahead and make that tweak then!